framework/db/ActiveQuery.php

```diff
@@ -382,8 +396,42 @@ protected function queryScalar($selectExpression, $db)
      */
     public function joinWith($with, $eagerLoading = true, $joinType = 'LEFT JOIN')
     {
-        $this->joinWith[] = [(array) $with, $eagerLoading, $joinType];
+        $relations = [];
+        foreach((array) $with as $name => $callback) {
+            if (is_int($name)) {
+                $name = $callback;
+                $callback = null;
+            }
 
+            if (preg_match('/^(.*?)(\s+AS\s+|\s+)(\w+)$/i', $name, $matches)) {
+                // relation is defined with an alias, adjust callback to apply alias
+                list(, $relation, , $alias) = $matches;
```

Comment:
In the regex above why doing second match if it's not used? Could be non-match group instead: `(?:\s+AS\s+|\s+)`